### PR TITLE
Leslie:  Week 3 homework pull request

### PR DESCRIPTION
**Pull request by noon-coffee**

### CODE REVIEW

File: daos/book.js [hunk 1]
```diff
@@ -4,8 +4,73 @@ const Book = require('../models/book');
 
 module.exports = {};
 
-module.exports.getAll = (page, perPage) => {
-  return Book.find().limit(perPage).skip(perPage*page).lean();
+module.exports.getAll = (page, perPage, authorId) => {
+  //return Book.find().limit(perPage).skip(perPage*page).lean();
+
+  //Include filter for authorId if specified and modified to use aggregate query
+  let findFilter = authorId ? { authorId : authorId } : {}
```
Comment (t1mwillis) [inline daos/book.js:11]:
```suggestion
  const findFilter = authorId ? { authorId } : {}
```
curious if ES6 would work here

File: daos/book.js [hunk 2]
```diff
@@ -4,8 +4,73 @@ const Book = require('../models/book');
...
+      }},
+      { $lookup: {
+        from: "authors",
+        localField: "authorId",
+        foreignField: "_id",
+        as: "author"
+      }},
+      { $unwind: '$author' }
+    ]
+    : [ //includeAuthorInfo = false
```
Comment (t1mwillis) [inline daos/book.js:56]: We spoke about this in class, but this is just an array, so consider declaring the portion that is shared between both queries and then `array.push()`ing the other portion of the query if needed.

File: routes/books.js [hunk 3]
```diff
@@ -22,6 +22,27 @@ router.post("/", async (req, res, next) => {
...
+  }
+
+  const books = await bookDAO.search(query);
+  res.json(books);
+});
+
+// Retrieve author stats
+router.get("/authors/stats", async (req, res, next) => {
+  let {authorInfo} = req.query;
+  let includeAuthorInfo = authorInfo ? true : false;
```
Comment (t1mwillis) [inline routes/books.js:40]: if you won't be re-assigning the variable, make sure to use `const`.

Also, in this instance, `undefined` is falsy, so this check here makes this readable, but is not required.

**Comment by t1mwillis** [comment]
Great work! Be careful with `let/const`!
Component | Points Available | Points Received
-- | -- | --
All tests, as originally given, are passing. | 80 | 80
Clear, organized project structure | 20 | 20
Total | 100 | 100

File: routes/books.js (hunk 3 above)
Comment (noon-coffee) [reply routes/books.js:40]: Thanks for the timely feedback and course1 refresher, I changed all the ones I could find in the next homework to use "const" instead :)

File: daos/book.js (hunk 2 above)
Comment (noon-coffee) [reply daos/book.js:56]: Ah nice, makes sense.

**Comment by noon-coffee** [comment]
Homework completed and graded.

**Pull request closed without merging**